sqlest/src/main/scala/sqlest/ast/Column.scala

```diff
@@ -119,6 +119,41 @@ sealed trait AliasedColumn[A] extends Column[A] with CellExtractor[ResultSet, A]
 
 }
 
+/**
+ * A column that has an index associated with it.
+ *
+ * This column is only for internal use when accessing generated keys
+ *
+ */
+case class IndexedColumn[A](index: Int)(implicit val columnType: ColumnType[A]) extends Column[A] with CellExtractor[ResultSet, A] {
```

Comment:
Can you elaborate?  I couldn't see how else to create an extractor from a column type, and the existing columns access resultset by name, not by index.  By name access would force the dev to specify the column name rather than just get generated keys.